dev_guide/image_signatures.adoc

```diff
@@ -0,0 +1,192 @@
+====
+endif::[]
+ifdef::openshift-origin[]
+[NOTE]
+====
+Initial support for storing image signatures was added in {product-title} 1.3.
+====
+endif::[]
+
+
+[[signing-images-using-atomic-cli]]
+== Signing Images Using Atomic CLI
+
+The `atomic` command line interface (CLI), version 1.12.5 or greater, provides
+commands for signing container images, which can be pushed to an OpenShift
+Container Registry.
+
+You can sign an image at push time:
+
+----
+$ atomic push [--sign-by ...] <image>
+----
+
+Or you can sign an image that has already been pushed to the registry:
+
+----
+$ atomic sign [--sign-by ...] <image>
+----
+
+For full details on how to set up and perform image signing using the `atomic`
+CLI, see the
+link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html/managing_containers/signing_container_images[RHEL Atomic Host Managing Containers: Signing Container Images] documentation.
+
+A specific example workflow of working with the `atomic` CLI and an OpenShift
+Container Registry is documented in the
+link:https://access.redhat.com/articles/2750891#working-with-openshift-and-atomic-registry[Container Image Signing Integration Guide].
+
+[[accessing-image-signatures-using-registry-api]]
+== Accessing Image Signatures Using Registry API
+
+The OpenShift Container Registry provides an `extensions` endpoint that allows
+you to write and read image signatures. The image signatures are stored in the
+{product-title} key-value store via the Docker Registry API.
+
+[NOTE]
+====
+This endpoint is experimental and not supported by the upstream Docker Registry
+project. See the link:https://docs.docker.com/registry/spec/api/[upstream API
+documentation] for general information about the Docker Registry API.
+====
+
+[[writing-image-signatures-using-registry-api]]
+=== Writing Image Signatures
+
+In order to add a new signature to the image, you can use the HTTP `PUT` method
+to send a JSON payload to the `extensions` endpoint:
+
+----
+PUT /extensions/v2/<namespace>/<name>/signatures/<digest>
+----
+
+----
+$ curl -X PUT --data @signature.json http://<user>:<token>@<registry_endpoint>:5000/extensions/v2/<namespace>/<name>/signatures/sha256:<digest>
+----
+
+The JSON payload with the signature content should have the following structure:
+
+----
+{
+  "version": 2,
+  "type":    "atomic",
+  "name":    "sha256:4028782c08eae4a8c9a28bf661c0a8d1c2fc8e19dbaae2b018b21011197e1484@cddeb7006d914716e2728000746a0b23",
+  "content": "<base64_encoded_signature>"
+}
+----
+
+The `name` field contains the name of the image signature, which must be unique
+and in the format `<digest>@<name>`. The `<digest>` represents an image name and
+the `<name>` is the name of the signature. The signature name must be 32
+characters long.
+
+[NOTE]
+====
+In order to attach the signature to the image, the user must have the
+`system:image-signer` role in the relevant project.
```

Comment:
It looked to me like there was both (`oc adm` for cluster-scoped):

```
# oc policy add-role-to-user system:image-signer fred -n alexd-project
role "system:image-signer" added: "fred"

# oc adm policy add-cluster-role-to-user system:image-signer alexd
cluster role "system:image-signer" added: "alexd"
```

If this is accurate, I figured I would add info on the cluster-scoped role to:

https://docs.openshift.com/container-platform/3.5/install_config/registry/accessing_registry.html#access-user-prerequisites

^ Similar to the note in there about the `system:image-builder` role.